download-artifacts/README.md

```diff
@@ -0,0 +1,44 @@
+# download-artifacts
+
+This action lets a workflow download artifacts from a job. An optional
+pattern can be specified.
+
+## Usage
+
+To use this, add a workflow file such as:
+
+```yaml
+name: ci-test-complete
+
+on:
+  # This action is often appropriate for workflow_run jobs, but it can be used with any kind of job
+  workflow_run:
+    workflows: ["ci-test"]
+    types:
+      - completed
+
+jobs:
+  process-artifacts:
+    runs-on: ubuntu-latest
+    name: Process artifacts
+    steps:
+      - name: Download and extract artifacts
+        uses: diem/actions/download-artifacts@<version>
+        with:
+          # The ID for the workflow run.
+          run-id: ${{ github.event.workflow_run.id }}
+          # "pattern" specifies a regex of artifact names to download. Default is to download all artifacts.
+          pattern: "^.*-test-results$"
```

Comment:
may want to specify a java regex parser, vs shell, a meh nit.

Reply:
could you elaborate? this is using standard JS regexes.